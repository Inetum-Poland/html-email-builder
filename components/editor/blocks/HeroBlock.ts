import type { BlockProperties } from "grapesjs";
import { Megaphone } from "lucide-static";

export const HeroBlock = (): BlockProperties => {
  const { $i18n: { t } } = useNuxtApp();

  return {
    label: t("hero"),
    media: Megaphone,
    content: `
    <mj-hero mode="fixed-height" height="470px" background-width="600px" background-url="https://placehold.co/600x400?text=Background%20Image" padding="100px 0">
      <mj-text align="center" font-size="3rem" line-height="1.15">
        ${t("heading")}
      </mj-text>
      <mj-button href="#" align="center">
        ${t("callToAction")}
      </mj-button>
    </mj-hero>`,
    activate: true,
  };
};
