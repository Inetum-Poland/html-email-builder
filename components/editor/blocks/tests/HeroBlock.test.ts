import { expect, test } from "vitest";
import { HeroBlock } from "../HeroBlock";

test("renders the Hero block", () => {
  expect(HeroBlock()).toMatchSnapshot();
});
