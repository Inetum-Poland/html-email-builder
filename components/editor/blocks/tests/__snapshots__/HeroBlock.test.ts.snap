// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the Hero block 1`] = `
{
  "activate": true,
  "content": "
    <mj-hero mode="fixed-height" height="470px" background-width="600px" background-url="https://placehold.co/600x400?text=Background%20Image" padding="100px 0">
      <mj-text align="center" font-size="3rem" line-height="1.15">
        Heading
      </mj-text>
      <mj-button href="#" align="center">
        Call to action ‚ü∂
      </mj-button>
    </mj-hero>",
  "label": "Hero",
  "media": "
<svg
  class="lucide lucide-megaphone"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="m3 11 18-5v12L3 14v-3z" />
  <path d="M11.6 16.8a3 3 0 1 1-5.8-1.6" />
</svg>
",
}
`;
